pkg/csi/service/common/util.go

```diff
@@ -242,16 +241,18 @@ func ParseStorageClassParams(ctx context.Context, params map[string]string) (*St
 				param = strings.ToLower(param)
 				if param == DatastoreMigrationParam {
 					scParams.Datastore = value
-				} else if param != DiskFormatMigrationParam && param != HostFailuresToTolerateMigrationParam &&
-					param != ForceProvisioningMigrationParam && param != CacheReservationMigrationParam &&
-					param != DiskstripesMigrationParam && param != ObjectspacereservationMigrationParam &&
-					param != IopslimitMigrationParam {
-					return nil, fmt.Errorf("Invalid parameter key:%v, value:%v", param, value)
+				} else if param == DiskFormatMigrationParam || param == HostFailuresToTolerateMigrationParam ||
```

Comment:
How about defining a map of unsupported params by CSI and then check if param exists in the map?

Reply:
done

```
unSupportedParameters := parameterSet{
   DiskFormatMigrationParam:             struct{}{},
   HostFailuresToTolerateMigrationParam: struct{}{}, 
   ForceProvisioningMigrationParam:      struct{}{},
   CacheReservationMigrationParam:       struct{}{},
   DiskstripesMigrationParam:            struct{}{},
   ObjectspacereservationMigrationParam: struct{}{},
   IopslimitMigrationParam:              struct{}{},		
}
```

```
else if unSupportedParameters.Has(param) {
	=return nil, fmt.Errorf("vSphere CSI driver does not support creating volume using in-tree vSphere volume plugin parameter key:%v, value:%v", param, value)
} 
```

Reply:
TBH I think this is not required and the earlier code was just fine.

Reply:
reverted to earlier code